source/configure/file-storage-configuration-settings.rst

```diff
@@ -418,6 +418,65 @@ Enable Amazon S3 debugging
+  :description: Amount of time, in milliseconds, before requests to Amazon S3 time out. Default value is 30000 (30 seconds).
+
+Amazon S3 request timeout 
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
++---------------------------------------------------------------+-----------------------------------------------------------------------------------------+
+| The amount of time, in milliseconds, before requests to       | - System Config path: N/A                                                               |
+| Amazon S3 storage time out.                                   | - ``config.json`` setting: ``".FileSettings.AmazonS3RequestTimeoutMilliseconds: 30000`` |
+|                                                               | - Environment variable: ``MM_FILESETTINGS_AMAZONS3REQUESTTIMEOUTMILLISECONDS``          |
+| Default is 30000 (30 seconds).                                |                                                                                         |
++---------------------------------------------------------------+-----------------------------------------------------------------------------------------+
+
+.. config:setting:: file-amazons3uploadpartsizebytes
+  :displayname: Amazon S3 upload part size (File Storage)
+  :systemconsole: N/A
+  :configjson: .FileSettings.AmazonS3UploadPartSizeBytes
+  :environment: MM_FILESETTINGS_AMAZONS3UPLOADPARTSIZEBYTES
+  :description: The size, in bytes, of each part in a multi-part upload to Amazon S3. Default value is 5242880 (5MB).
+
+Amazon S3 upload part size
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++---------------------------------------------------------------+---------------------------------------------------------------------------------------+
+| The size, in bytes, of each part in a multi-part              | - System Config path: N/A                                                             |
+| upload to Amazon S3.                                          | - ``config.json`` setting: ``".FileSettings.AmazonS3UploadPartSizeBytes: 5242880``    |
+|                                                               | - Environment variable: ``MM_FILESETTINGS_AMAZONS3UPLOADPARTSIZEBYTES``               |
+| Numeric value. Default is 5242880 (5MB).                      |                                                                                       |
++---------------------------------------------------------------+---------------------------------------------------------------------------------------+
+| **Note**: A smaller part size results in more parts to manage, which can increase overhead while being more reliable for networks with less stable    |
+| connections. A larger part size reduces the number of upload parts, but requires more stable network conditions, and can increase the impact of any   |
+| failures requiring parts to be re-uploaded.                                                                                                           |
```

Comment:
This is technically correct but the way we do uploads means the tradeoff is more between memory (having bigger parts means more allocations) and time to upload (having smaller parts means more requests, hence latency).

Reply:
Thanks, @streamer45! I'll rework these notes accordingly.

Reply:
Thanks! :raised_hands:

Reply:
My latest commit incorporates your feedback, @streamer45.